config.template.yml

```diff
@@ -277,14 +277,22 @@ notifier:
   ## filesystem:
   ##   filename: /tmp/authelia/notification.txt
 
-  # Use a SMTP server for sending notifications. Authelia uses PLAIN method to authenticate.
-  # [Security] Make sure the connection is made over TLS otherwise your password will transit in plain text.
+  # Use a SMTP server for sending notifications. Authelia uses PLAIN or LOGIN method to authenticate.
+  # [Security] By default Authelia will:
+  #   - force all SMTP connections over TLS including unauthenticated connections
+  #      - use the disable_require_tls boolean value to disable this requirement (only works for unauthenticated connections)
+  #   - validate the SMTP server x509 certificate during the TLS handshake against the hosts trusted certificates
+  #     - use the trusted_cert string value to specify the path of a PEM format public cert to trust in addition to the hosts trusted certificates
+  #     - use the disable_verify_cert boolean value to disable the validation (prefer the trusted_cert option as it's more secure)
   smtp:
     username: test
     password: password
     host: 127.0.0.1
     port: 1025
     sender: admin@example.com
+    ## disable_require_tls: False
+    ## disable_verify_cert: False
+    ## trusted_cert: ""
```

Comment:
Can you have the commented `trusted_cert:` set to `/etc/sl/certs/ca-certificates.crt` because this is the default CA cert within the Docker container.

Reply:
So the way I've implemented trusted_cert this is not entirely necessary. It loads the original certificate pool which will be from that location, and adds the certificate from the location specified in trusted_cert as an additional trusted_cert.

So with it blank, commented, or invalid, it will load /etc/ssl/certs/ca-certificates.crt because of go's method to load the system cert pool: https://golang.org/src/crypto/x509/cert_pool.go?s=1351:1391#L45

Do you still want me to change it?

Reply:
That's actually pretty cool, no I don't think there's a requirement to change the `trusted_cert` anymore, but updating documentation or documenting that feature would be useful so other people know too.

Reply:
I'll update the documentation. It is already somewhat documented but I'll figure out how to make it clearer: https://github.com/james-d-elliott/authelia/blob/feature/notifier-smtp-security-improvements/config.template.yml#L285

Reply:
I think the documentation changes should resolve this.

Reply:
Yep thanks for that, LGTM!